modules/Classroom/components/CreateAssignmentFormModal/CreateAssignmentFormModal.tsx

```diff
@@ -16,55 +16,92 @@ import { DateInput } from "@mantine/dates";
-import { ICreateAssignmentFormModalProps } from "./CreateAssignmentFormModal.interface";
+import {
+  assignmentFormSchema,
+  assignmentFormSchemaEdit,
+  TAssignmentFormValues,
+} from "./CreateAssignmentFormModal.helpers";
+import { IAssignmentFormModalProps } from "./CreateAssignmentFormModal.interface";
 import { formStyles, inputStyles } from "./CreateAssignmentFormModal.styles";
 
-const CreateAssignmentFormModal: React.FC<ICreateAssignmentFormModalProps> = ({
+const CreateAssignmentFormModal: React.FC<IAssignmentFormModalProps> = ({
   opened,
   onClose,
   classroomId,
+  assignment,
 }) => {
   const {
     control,
     handleSubmit,
     register,
     formState: { errors },
     reset,
+    setValue,
   } = useForm<TAssignmentFormValues>({
-    resolver: zodResolver(assignmentFormSchema),
+    resolver: zodResolver(assignment ? assignmentFormSchemaEdit : assignmentFormSchema),
     mode: "onBlur",
   });
 
   const [isLoading, setIsLoading] = useState(false);
   const [createAssignment] = useCreateAssignmentMutation();
+  const [updateAssignment] = useUpdateAssignmentMutation();
+
+  useEffect(() => {
+    if (assignment) {
+      setValue("title", assignment.title);
+      setValue("description", assignment.description);
+      setValue("dueDate", new Date(assignment.dueDate));
+    }
+  }, [assignment, setValue]);
 
   const onSubmit = async (data: TAssignmentFormValues) => {
     setIsLoading(true);
     try {
-      await createAssignment({
-        classroomId,
-        data: {
-          file: data.file,
-          title: data.title,
-          description: data.description,
-          dueDate: data.dueDate,
-        },
-      }).unwrap();
+      if (assignment) {
+        await updateAssignment({
+          assignmentId: assignment.id,
+          data: {
+            file: data.file,
+            title: data.title,
+            description: data.description,
+            dueDate: data.dueDate,
+          },
+        }).unwrap();
+
+        showNotification({
+          title: "Success",
+          message: "Assignment updated successfully!",
+          color: "blue",
+        });
+      } else {
+        await createAssignment({
+          classroomId,
+          data: {
+            file: data.file,
+            title: data.title,
+            description: data.description,
+            dueDate: data.dueDate,
+          },
+        }).unwrap();
+
+        showNotification({
+          title: "Success",
+          message: "Assignment created successfully!",
+          color: "blue",
+        });
+      }
 
-      showNotification({
-        title: "Success",
-        message: "Assignment created successfully!",
-        color: "blue",
-      });
       reset();
       onClose();
     } catch (error) {
       showNotification({
-        title: "Assignment creation failed",
-        message: "Failed to create assignment",
+        title: assignment ? "Update failed" : "Creation failed",
```

Comment:
Does it make sense to track a boolean var named something like `isUpdateAssignment` for these checks?

It is a bit confusing when you are checking if assignment exists in the catch block.